src/freenas/usr/local/libexec/freenas-debug/smart/smart.sh

```diff
@@ -92,10 +92,30 @@ smart_func()
 		disks=$(sysctl -n kern.disks)
 	fi
 
+	# SAS to SATA interposers could be involed. Unfortunately,
+	# there is no "easy" way of identifying that there is
+	# one involved without doing some extravagant reading of
+	# specific VPD pages from the device itself. Even doing
+	# that is fraught with errors because the interposer could
+	# mistranslate those pages.
+	# So, instead, we'll try to tell smartctl to do the translation
+	# and if it fails (which it will on proper SAS devices) then
+	# we'll try to run it without translation
 	for i in $disks
 	do
-    		echo /dev/$i >> /tmp/smart.out
-		smartctl -a /dev/$i >> /tmp/smart.out
+		# try with translation first
+		output=$(smartctl -a -d sat /dev/$i)
```

Comment:
What happen if `-d sat` is used with non-SCSI disk like SATA or NVMe?  Probably nothing bad, but still.  Also I worry about using it on USB sticks, since I remember some ugly USB devices hanging from unexpected commands.

Reply:
In my testing `-a -d sat /dev/*` on a SATA HDD/SSD completed without problems. It failed when talking to SAS or interposer device for which `-a /dev/*` worked.

To answer your question about NVMe, it seems that's never worked quite right because `sysctl -n kern.disks` is returning `nvd*` devices so `-a -d sat /dev/nvd*` or `-a /dev/nvd*` fails because it's expecting the `/dev/nvme*` device. So that will need to be fixed.